main.cpp

```diff
@@ -133,256 +143,329 @@ struct CarWash
 };
 
+bool connectToInternet(bool paidISP = true)
+{
+    return paidISP ? true : false;
+}
+
+bool save(std::string fileName, bool clickSaveButton)
+{
+    return fileName != "" && clickSaveButton ? true : false;
+}
+
+bool runApp(bool doubleClick)
+{
+    return doubleClick ? true : false;
+}
+
 struct Car
 {
-    // engine size
     float engineSize = 13.1f;
-    // make
     std::string make = "Subaru";
-    // model
     std::string model = "Outback";
-    // number of seats
     int numberOfSeats = 5;
-    // average miles per gallon (float)
     float averageMPG = 30.2f;
 
-    // 3 things it can do:
-    // 1) the car can move forward
-    int moveForward(int distanceTraveled, int time); // returns speed
-    // 2) the car can stop
-    void stopMovement();
-    // 3) the car can turn
-    bool turn(); // return whether the car will turn or not
+    int moveForward(int distanceTraveled, int time);
+    bool stopMovement(bool applyBrakes);
+    bool turn(bool rotateSteeringWheel);
     
 };
 
-struct Tree
+int moveForward(int distanceTraveled, int time)
 {
-    // height 
+    return distanceTraveled / time;
+}
+
+bool stopMovement(bool applyBrakes)
+{
+    return applyBrakes ? true : false;
+}
+
+bool turn(bool rotateSteeringWheel)
+{
+   return rotateSteeringWheel ? true : false; 
+}
+
+struct Tree
+{ 
     float height = 13.7f;
-    // age
     int age = 23;
-    // type 
     std::string type = "Apple";
-    // location
     std::string location = "Backyard";
-    // number of branches (int)
     int numberOfBranches = 10349;
 
-    // Nested UDT
     struct Fruit
     {
-        // member variables
-        std::string name= "Apple";
+        std::string name = "Apple";
         std::string color = "Green";
         bool readyToEat = false;
         int numberOfSeeds = 5;
-        // member functions
-        int produceSeeds(); // returns number of seeds produced
+
+        int produceSeeds();
         void fallFromTree();
-        void inspireTheoryOfGravity(); // did this actually happen with Newton?
+        void inspireTheoryOfGravity();
     };
-    // 3 things it can do:
-    // 1) grow
-    int grow(int sunlight, int water); // returns growth amount
-    // 2) produce fruit
+
+    int grow(int sunLight, int water);
     Fruit produceFruit();
-    // 3) photosynthesize
-    void photosynthesize();
+    int photosynthesize(int sunLight, int water, int unkonwnFactor);
 };
 
-struct Person
+int grow(int sunLight, int water)
+{
+    return sunLight * water;
+}
+// I couldn't quite figure out how to make this return a Fruit 
+// Tree::Fruit produceFruit(int grow)
+// {
+//      if (grow > 0)
+//      {
+//          return
+//             bool readyToEat = true;
+//      }
+//     else 
+//      {      
+//         return
+//             bool readyToEat = false;
+//     }
+// }
+
+int photosynthesize(int sunLight, int water, int unkonwnFactor)
+{
+    return (sunLight > 0) ? water * unkonwnFactor : 0;
+}
```

Comment:
These are not member functions.  these are free functions.
You did not fully qualify the function name
- [ ] revise